src/common/server.c

```diff
@@ -200,13 +200,35 @@ tcp_send_len (server *serv, char *buf, int len)
 	}
 	else
 	{
-		/* WHO/MODE get the lowest priority */
-		if (g_ascii_strncasecmp (dbuf + 1, "WHO ", 4) == 0 ||
-		/* but only MODE queries, not changes */
-			(g_ascii_strncasecmp (dbuf + 1, "MODE", 4) == 0 &&
-			 strchr (dbuf, '-') == NULL &&
-			 strchr (dbuf, '+') == NULL))
+		/* WHO gets the lowest priority */
+		if (g_ascii_strncasecmp (dbuf + 1, "WHO ", 4) == 0)
 			dbuf[0] = 0;
+		/* as do MODE queries (but not changes) */
+		else if (g_ascii_strncasecmp (dbuf + 1, "MODE", 4) == 0)
+		{
+			char *mode_str, *mode_str_end, *loc;
+			/* skip spaces before channel/nickname */
+			for (mode_str = dbuf + 5; *mode_str == ' '; ++mode_str);
```

Comment:
Is it possible for an empty `"MODE\0"` to make it here?

Reply:
That's a good question. It's not a valid command (it must always have at least one argument as the user/channel), so we shouldn't be generating it, but I can imagine there's nothing stopping something else from doing it. I suppose we could "cheat" and do `g_ascii_strncasecmp(dbuf + 1, "MODE ", 5)` like with `"WHO "`?

Reply:
Yes that sounds fine.

Reply:
Forgot to ever get back to this; I have just pushed a fixed version. Having said that, I don't think it would actually have mattered; `MODE\0` would end up with `mode_str` pointing at the `\0`, and `strchr` would then return `NULL`. The main difference is if you ever issue a `MODEFOO` command, which previously (both before this patch and with the previous version of this patch) would have a lower priority. Also we now don't give a `MODE\0` command low priority, which we previously would have done.